Test/Unit/Context.php

```diff
@@ -0,0 +1,224 @@
+                    ->isInstanceOf('Hoa\Math\Exception\UnknownConstant');
+    }
+
+    public function case_context_exception_when_setting_already_defined_constant()
+    {
+        $this
+            ->given(
+                $name    = 'PI',
+                $context = new CUT()
+            )
+            ->then
+                ->exception(function () use ($context, $name) {
+                    $context->addConstant($name, 42);
+                })
+                    ->isInstanceOf('Hoa\Math\Exception\AlreadyDefinedConstant');
+    }
+
+    public function case_context_returns_constant_value()
+    {
+        $this
+            ->given(
+                $name     = 'FOO',
+                $value    = 42,
+                $context  = new CUT()
+            )
+            ->when($context->addConstant($name, $value))
+            ->then
+                ->variable($context->getConstant($name))->isEqualTo($value);
+    }
+
+    public function case_context_has_predefined_functions()
+    {
+        $this
+            ->given($context = new CUT())
+            ->then
+                ->array(iterator_to_array($context->getFunctions()))
+                    ->hasSize(22)
+                    ->hasKey('abs')
+                    ->hasKey('acos')
+                    ->hasKey('asin')
+                    ->hasKey('atan')
+                    ->hasKey('average')
+                    ->hasKey('avg')
+                    ->hasKey('ceil')
+                    ->hasKey('cos')
+                    ->hasKey('count')
+                    ->hasKey('deg2rad')
+                    ->hasKey('exp')
+                    ->hasKey('floor')
+                    ->hasKey('ln')
+                    ->hasKey('log')
+                    ->hasKey('max')
+                    ->hasKey('min')
+                    ->hasKey('pow')
+                    ->hasKey('rad2deg')
+                    ->hasKey('sin')
+                    ->hasKey('sqrt')
+                    ->hasKey('sum')
+                    ->hasKey('tan');
+    }
+
+    public function case_context_exception_when_getting_unknown_function()
+    {
+        $this
+            ->given(
+                $name    = 'foo',
+                $context = new CUT()
+            )
+            ->then
+                ->exception(function () use ($context, $name) {
+                    $context->getFunction($name);
+                })
+                    ->isInstanceOf('Hoa\Math\Exception\UnknownFunction');
+    }
+
+    public function case_context_exception_when_setting_unknown_function()
+    {
+        $this
+            ->given(
+                $name    = 'foo',
+                $context = new CUT()
+            )
+            ->then
+                ->exception(function () use ($context, $name) {
+                    $context->addFunction($name);
+                })
+                    ->isInstanceOf('Hoa\Math\Exception\UnknownFunction');
+    }
+
+    public function case_context_returns_function_callable()
```

Comment:
What do you test with this kind of test case? The `add*` method or the `get*` method? If both, please, split in 2 distinct test cases.